### ISSUE TITLE
Remove unused CSS

### ISSUE THREAD

**Issue by stevenvachon** [posthtml/htmlnano#36]
with [email-remove-unused-css](https://npmjs.com/email-remove-unused-css).

It's built with posthtml, but it currently cannot be used as a plugin and forces a reparse of the HTML. This should [get fixed](https://github.com/code-and-send/email-remove-unused-css/issues/1), though.

**Comment by revelt** [comment]
I agree, the unused CSS removal is a perfect feature for minifier.

And there can be issues when any removal of CSS is done _after_ the minification because, for example, mentioned [email-remove-unused-css](https://npmjs.com/email-remove-unused-css) parses and renders back the code. The CSS renderer (stringifier) won't output minified code, what means minifying would need to be performed **again**, OR unused CSS removed first and only then whole thing minified. When minification does the CSS removal it's faster for user.

[email-remove-unused-css](https://npmjs.com/email-remove-unused-css) is for email code and was not designed to accept **external stylesheets**. That will be an issue for `htmlnano` which is _web pages_ oriented. Unless we're happy with removal only what's contained. In a sense that's fine since it's quite quick procedure, it's not unCSS.

There is a PostHTML plugin to remove unused CSS, https://www.npmjs.com/package/postcss-remove-unused and looks like it accepts external stylesheets, although probably not `id` attributes.

**Comment by fstanis** [comment]
How about using [uncss](https://github.com/uncss/uncss)?

It seems that, at least theoretically, we could add uncss to the `minifyCss` module with the help of `posthtml-render` (since uncss requires the html):

```js
const uncss = require('uncss');
const render = require('posthtml-render');

[...]

function processStyleNode(tree, styleNode, cssnanoOptions) {
  const css = Array.isArray(styleNode.content) ? styleNode.content.join(' ') : styleNode.content;
  [...]
  const html = render(tree, options);
  uncss(html, { stylesheets: [], raw: css } (error, output) => { ... });
}
```

Thoughts on this?

**Comment by maltsev** [comment]
Thanks, @fstanis for implementing that! I released this module and your other fixes as [version 0.2.2](https://github.com/posthtml/htmlnano/releases/tag/0.2.2).
